pontoon/base/placeables.py

```diff
@@ -0,0 +1,64 @@
+import re
+
+
+class MozillaPrintfString:
+    parser = re.compile(r"%(?:[1-9]\$)?[S@]")
+
+
+class PythonPrintfString:
+    parser = re.compile(
+        r"""
+            %
+            (?:\(.*?\))?  # mapping key
+            [\#0\-\ +]*   # conversion flags
+            [\d*]*        # minimum field width
+            (?:\.[\d*])?  # precision
+            [hlL]?        # length modifier
+            [acdEeFfGgiorsuXx%]  # conversion type
+        """,
+        re.X,
+    )
+
+
+class PythonFormatString:
+    parser = re.compile(r"{{|}}|{[\w!.,[\]%:$<>+\-= ]*}")
+
+
+class FluentTerm:
+    parser = re.compile(r"({ *-[^}]*})")
+
+
+class FluentFunction:
+    parser = re.compile(r"({ *[A-W0-9\-_]+[^}]*})")
+
+
+class JsonPlaceholder:
+    parser = re.compile(r"(\$[A-Z0-9_]+\$)")
+
+
+def get_placeables(text):
+    """Return a list of placeables found in the given string."""
+    parsers = [
+        MozillaPrintfString.parser,
+        PythonPrintfString.parser,
+        PythonFormatString.parser,
+        FluentTerm.parser,
+        FluentFunction.parser,
+        JsonPlaceholder.parser,
```

Comment:
```suggestion
        JsonPlaceholder.parser,
        MozillaPrintfString.parser,
        PythonPrintfString.parser,
        PythonFormatString.parser,
        FluentPlaceable.parser,
```

Merge the Fluent parsers, try `JsonPlaceholder` first so that it'll capture e.g. `$S$` correctly.